### PR TITLE
[docs] Add cmake flags that reduce memory usage

### PR DESCRIPTION
**Pull request by dobios**
This PR introduces some additional cmake flags to the GettingStarted docs that help reduce memory usage.

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -269,4 +282,4 @@ Patches are submitted to LLVM/MLIR via GitHub pull-requests, the basic flow is a
 6) Publish the branch on your fork of the repository and create a GitHub pull-request.
 
 When your review converges and your patch is approved, it can be merged directly on GitHub.
-If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.
+If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.
```
Comment (fabianschuiki) [inline docs/GettingStarted.md:285]: Stray whitespace :monocle_face:

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -81,11 +86,17 @@ $ cmake -G Ninja .. \
...
+    -DLLVM_ENABLE_LLD=ON  
 $ ninja
 $ ninja check-circt
 $ ninja check-circt-integration # Run the integration tests.
-```
+```  
+In order to use the recommended `lld` linker, use the `-DLLVM_ENABLE_LLD=ON`. 
+Removing that flag will use your compiler's default linker. More details about
+these problems and their solutions can be found 
+[in the llvm docs](https://llvm.org/docs/GettingStarted.html#common-problems).
```
Comment (fabianschuiki) [inline docs/GettingStarted.md:99]:
```suggestion
[in the LLVM docs](https://llvm.org/docs/GettingStarted.html#common-problems).  
```

**Review by fabianschuiki — APPROVED** [review]
Thanks for adding this!

**Review by teqdruid — APPROVED** [review]
Thanks! Fabian's comments excepted, this looks gtg!

**Pull request merged**